duprequests/middleware.py

```diff
@@ -1,41 +1,57 @@
-
-from uuid import uuid4
+from uuid import uuid4, uuid5, NAMESPACE_DNS
+from urllib.parse import urlencode
 
 from django.conf import settings
 from django.core.cache import caches
 from django.http.response import HttpResponseNotModified
 
+
 try:
     from django.utils.deprecation import MiddlewareMixin
-except ImportError:   # pragma: nocover
+except ImportError:  # pragma: nocover
     MiddlewareMixin = object
 
-CACHE_NAME = getattr(settings, 'DUPLICATED_REQUESTS_CACHE_NAME', 'default')
-CACHE_TIMEOUT = getattr(settings, 'DUPLICATED_REQUESTS_CACHE_TIMEOUT', 5)
-COOKIE_NAME = getattr(settings, 'DUPLICATED_REQUESTS_COOKIE_NAME',
-                      'dj-request-id')
-COOKIE_PREFIX = getattr(settings, 'DUPLICATED_REQUESTS_COOKIE_PREFIX',
-                        'request-id-')
+CACHE_NAME = getattr(settings, "DUPLICATED_REQUESTS_CACHE_NAME", "default")
+CACHE_TIMEOUT = getattr(settings, "DUPLICATED_REQUESTS_CACHE_TIMEOUT", 5)
+COOKIE_NAME = getattr(settings, "DUPLICATED_REQUESTS_COOKIE_NAME", "dj-request-id")
+COOKIE_PREFIX = getattr(settings, "DUPLICATED_REQUESTS_COOKIE_PREFIX", "request-id-")
 
 
 class DropDuplicatedRequests(MiddlewareMixin):
-    """Middleware that drops requests made in quick succession.
-
-    Uses Django's caching system to check/save each request."""
+    """
+    Middleware that drops requests made in quick succession.
+    Uses Django's caching system to check/save each request.
+    """
+
+    @staticmethod
+    def _get_cookie_name(request):
+        hash_value = uuid5(
+            NAMESPACE_DNS,
+            request.path
+            + request.method.lower()
+            + urlencode(request.GET)
+            + "--"
+            + urlencode(request.POST),
+        ).node
+        return f"{COOKIE_NAME}-{hash_value}"
 
     def process_request(self, request):
-        if not request.method.lower() in ('post', 'put', 'delete', 'patch'):
+        if not request.method.lower() in ("post", "put", "delete", "patch"):
             return
 
-        cache_key = request.COOKIES.get(COOKIE_NAME)
-        if not cache_key:
+        cookie_name = self._get_cookie_name(request)
+        cookie_value = request.COOKIES.get(cookie_name)
+        if not cookie_value:
             return
 
+        cache_key = cookie_name + cookie_value
```

Comment:
Since cache key uses cookie_value and cookie value is changed after the `process_response`, the `HttpResponseNotModified` will never be raised.

As suggestion you could move `cache.set` to line 57